Office Hours 2019-09-03 (#18)
osquery/foundation: docs/office-hours/20190903_meeting_notes.md

# osquery Office Hours 2019-09-03

> 

Author: ```suggestion

YouTube Recording: https://youtu.be/urMFR1ICFbU
```

An example:
* augeas needs `fnmatch`
* The `fnmatch` in the older libc is buggy
* So augeas has a `posix_fnmatch`
* But autoconf won't detect it when it's running on a newer build system -- autoconf will make the best decision for the running environment, we need it to make the most portable decision.

Discussion about shipping a libc? Conclusion is that this isn't something worth pursuing.

There are some concerns about merging the toolchain work. As it stands, it probably breaks build and runtime on some older platforms. But, the current build uses the OS toolchain, which is also broken on those platforms. So merging is not worse.

What if we limited the supported platforms? Would this let us ship faster? What would our messaging here be? Does anyone use CentOS6. We believe that we are as stable as in prior releases. But we understand more about third party libraries, and we think there are places we can improve for older versions of libc.

More people to help update third-party libs would help. Stefano has done about seven libraries. We have about 40. Not all libraries are equal.

#### How do we move forward?

> We **will** merge the custom toolchain work. This needs a small amount of doc work, and binaries somewhere. https://github.com/osquery/osquer

Author: ```suggestion
We **will** merge the custom toolchain work. This needs a small amount of doc work, and binaries somewhere. https://github.com/osquery/osquery/issues/5609
```

> _Then_ we will tackle the third-party dependencies one at a time, getting support for older libc into their configuration files. This work will be done a third-party lib at a time.

Author: Do folks want me to make tickets for each third party lib? I can automate it, it's easy, but it is 40 new tickets

Reviewer: My take is that if we expect people to subdivide work, and do this gradually, I would say that we need them to be separate.
At the same time, we should though have a single issue which covers CentOS 6 support (or similar) that collects them all, with a check list for each of them.
Much like https://github.com/osquery/osquery/issues/5652.

Author: Works for me. https://github.com/osquery/osquery/issues/5745

> _Then_ we will tackle the third-party dependencies one at a time, getting support for older libc into their configuration files. This work will be done a third-party lib at a time.

Author: ```suggestion
_Then_ we will tackle the third-party dependencies one at a time, getting support for older libc into their configuration files. This work will be done a third-party lib at a time. https://github.com/osquery/osquery/issues/5745
```

This is seen as a larger topic for another time.

## Roadmap

### 4.0.2 -- Extension Support

https://github.com/osquery/osquery/issues/5636

The main issue is that to ship extension, we need the custom tool chain and dependencies. So extension support was pushed forward to 4.0.2.

### Future work

We should have a roadmap. Some comments:

* The metaprogramming around the spec files https://github.com/osquery/osquery/issues/5502
* Stronger typing
* A bunch of work from @packetzero

How do we talk about this? @mike-myers-tob suggests we use the GitHub project kanban tool for this. It's somewhere to center a discussion. And it's something supported directly in GitHub. We could use multiple boards, reflecting different time periods.

Let's try it. @directionless will click some buttons to set it up.

## Finance / Fundraising

This is https://github.com/osquery/foundation/issues/15

The readthedocs people reached out to us about profit sharing, which means we need a bank account. The Linux Foundation pointed us at Community Bridge, a platform they're building out.

> seph went through the signup procress, and we are now live at https://funding.communitybridge.org/projects/osquery It all looks a bit rough right now. But, we can probably take donations in, and the TSC can claim expenses.

Reviewer 2: ```suggestion
seph went through the signup process, and we are now live at https://funding.communitybridge.org/projects/osquery It all looks a bit rough right now. But, we can probably take donations in, and the TSC can claim expenses.
```
